suite_test.go

```diff
@@ -0,0 +1,119 @@
+// Licensed under the MIT license, see LICENCE file for details.
+
+package quicktest_test
+
+import (
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+)
+
+func TestRunSuite(t *testing.T) {
+	var calls []call
+	tt := &testingT{}
+	c := qt.New(tt)
+	c.RunSuite(testSuite{calls: &calls})
+	if len(calls) != 2 {
+		t.Fatalf("unexpected number of calls made: expected 2, got %d", len(calls))
+	}
+	if calls[0].Name != "Test1" {
+		t.Errorf("unexpected call[0]: expected Test1, got %s", calls[0].Name)
+	}
+	if calls[0].Init != false {
+		t.Error("unexpected call[0]: not initialized")
+	}
+	if calls[1].Name != "Test4" {
+		t.Errorf("unexpected call[1]: expected Test4, got %s", calls[1].Name)
+	}
+	if calls[1].Init != false {
+		t.Error("unexpected call[1]: initialized")
+	}
+}
+
+func TestRunSuitePtr(t *testing.T) {
+	var calls []call
+	tt := &testingT{}
+	c := qt.New(tt)
+	c.RunSuite(&testSuite{calls: &calls})
+	if len(calls) != 4 {
+		t.Fatalf("unexpected number of calls made: expected 4, got %d", len(calls))
+	}
+	if calls[0].Name != "Init" {
+		t.Errorf("unexpected call[0]: expected Init, got %s", calls[0].Name)
+	}
+	if calls[0].Init != false {
+		t.Error("unexpected call[0]: already initialized")
+	}
+	if calls[1].Name != "Test1" {
+		t.Errorf("unexpected call[1]: expected Test1, got %s", calls[1].Name)
+	}
+	if calls[1].Init != true {
+		t.Error("unexpected call[1]: not initialized")
+	}
+	if calls[2].Name != "Init" {
+		t.Errorf("unexpected call[2]: expected Init, got %s", calls[2].Name)
+	}
+	if calls[2].Init != false {
+		t.Error("unexpected call[2]: already initialized")
+	}
+	if calls[3].Name != "Test4" {
+		t.Errorf("unexpected call[3]: expected Test4, got %s", calls[3].Name)
+	}
+	if calls[3].Init != true {
+		t.Error("unexpected call[3]: not initialized")
+	}
+}
+
+type testSuite struct {
+	init  bool
+	calls *[]call
+}
```

Comment:
func (s *testSuite) addCall(name string) {
        *s.calls = append(*s.calls, call{Name: name, Init: s.init})
    }

?
Then use it in each method?